src/main/java/ch/njol/skript/sections/SecOpenInventory.java

```diff
@@ -0,0 +1,120 @@
+package ch.njol.skript.sections;
+
+import java.util.List;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.Inventory;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.effects.EffOpenInventory;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.Section;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Trigger;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+
+@Name("Open/Close Inventory Section")
+@Description({
+	"Opens an inventory to a player.",
+	"The section then allows to modify the event-inventory."
+})
+@Examples({
+	"show chest inventory to player:",
+		"\tset slot 1 of event-inventory to stone named \"example\"",
+		"\topen event-inventory to all players"
+})
+@Since("INSERT VERSION")
+public class SecOpenInventory extends Section {
```

Comment:
wouldn't it be easier to have this be an EffectSection, then EffOpenInventory can become just closing inventories and all the inventory logic is concentrated in this class? The this could support anvils and such as well.